CONTRIBUTING.rst

```diff
@@ -35,14 +35,15 @@ and will typically be labeled during triage with ``needs design``.
 
 After you have agreed upon an implementation strategy for your feature or patch with an MLflow
 committer, the next step is to introduce your changes (see `developing changes
-<https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#developing-and-testing-changes-to-mlflow>`_)
+<https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#developing-and-testing-changes-for-mlflow>`_)
```

Comment:
If we stick with the original wording per https://github.com/mlflow/mlflow/pull/3219/files#r472528940, we should undo this change

Reply:
Given that we're ok with the wording "Developing and testing MLflow" as the header, can we fix this link? I think it should be https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#developing-and-testing-mlflow, to match https://github.com/mlflow/mlflow/blob/ebacfdaf0095bf816a19f24a64319d6570d81f26/CONTRIBUTING.rst#developing-and-testing-mlflow